applications/nrf_desktop/src/modules/hid_state.c

```diff
@@ -1017,7 +1017,6 @@ static void connect(const void *subscriber_id, uint8_t report_id)
 		int8_t cur_sub_prio = get_subscriber_priority(rd->linked_rs->subscriber);
 		int8_t new_sub_prio = get_subscriber_priority(subscriber);
 
-		__ASSERT_NO_MSG(cur_sub_prio != new_sub_prio);
```

Comment:
what happened?

Reply:
In case we switch between protocol modes we may e.g. enable mouse report before we disable boot mouse report. In that case we get the same subscriber priority here (we actually refer to the same subscriber twice). This case lasts only for a brief moment (boot reports would be disabled by subsequent event related to HID subscription). Still it's enough to trigger assertion fail.